### ISSUE TITLE
Some method of properly passing Cloudflare protections?

### ISSUE THREAD

**Issue by Kadigan** [tenox7/wrp#110]
Hey.

So Cloudflare is everywhere now (enough to bring Discord down when they furk up), and it seems WRP doesn't know how to work with it. I can't seem to be able to pass Cloudflare's "verify you are human".

Any suggestions?

**Comment by TheTechRobo** [comment]
Maybe cloudflare is detecting the headless Chrome?

**Comment by Kadigan** [comment]
The particular website is actually known for being very aggressive about it (or maybe Cloudflare is particularly aggressive about that website? who knows). All I know is that I see 30-50 "checking the security of the connection" pages on an average day (and some days - at every page load), using a regular browser. I don't think hiding the headless Chrome better will fix it.

**Comment by TheTechRobo** [comment]
Headless Chrome is often used for data scraping. Websites like Cloudflare block it more aggressively. 

What exactly happens when you go to the page? Is there an "I'm not a robot" checkbox?

**Comment by Kadigan** [comment]
@TheTechRobo Yes, there is. It does the whole song & dance, and then redirects. Assuming the check works, the "browser" would need to be able to navigate, and/or send screen updates or something.

**Comment by tenox7** [comment]
do you have any handy examples of pages that fail this?

**Comment by Kadigan** [comment]
Anything on **fanfiction.net** should qualify. They pursue verification so aggressively, that I have to go through the verification almost every other chapter _on my desktop_.

**Comment by tenox7** [comment]
🤷 I have released 4.6.2, it has some basic anti crawler detection bits, also allows setting user agent with a flag; user agent seems to help the most;

I have tried `./wrp -ua="Mozilla/5.0 (Macintosh; Intel Mac OS X 10_15_7) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/119.0.0.0 Safari/537.36"` and it seems works most of the time?
